blocks/adaptors/base.py

```diff
@@ -23,3 +17,6 @@ def is_fitted(self) -> bool:
 
     def children(self) -> List[Element]:
         return [self]
+
+    def dict_children(self) -> dict:
+        return {"name": self.id, "obj": self}
```

Comment:
why is this a dict and not an object with properties? what's the advantage of a dict? looks like then you need random strings to access properties.

Reply:
It could be an object, but this way its an nested dictionary all the way down. What is the advantage of creating an object to accommodate for that?

Reply:
it can be a nested object all the way down, no? it's equivalent.
the advantage over a dictionary is type safety.

Reply:
I have PTSD from dictionaries, so happy that we don't use them anymore! it was so painful. but if you'd like this to be a dicitionary, sure. can you explain why?

Reply:
Just seems like a use case that isn't critical for type safety and we don't have to import a class type in 12 different files. + we don't have to maintain a separate class for it.

Reply:
ok! but a dictionary with set amount of properties that you need to remember every time you access it is an implicit dataclass.
is it easier to maintain an implicit dataclass or an explicit dataclass? is there a reason why an implicit one is easier to maintain?

Reply:
no use case is critical for type safety - it all adds up :)